### PR TITLE
Feature - Custom floating button

### PR DESCRIPTION
**Pull request by at-haingo**
#### Related issues (Optional)
Task #16

#### Description (Required)
- [x] Create view group contain floating button and text
- [x] Show title when touch to button
- [x] Animations

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+     */
+    internal fun setOnMenuItemClickListener(listener: OnMenuClickListener) {
+        mOnMenuClickListener = listener
+    }
+
+    /**
+     * Interface menu click listener
+     */
+    interface OnMenuClickListener {
+        fun onShareClick()
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        fun onProfileClick()
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        fun onCalendarClick()
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt [hunk 4]
```diff
@@ -0,0 +1,97 @@
+package vn.asiantech.way.ui.views
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.view.animation.Animation
+import android.view.animation.AnimationUtils
+import android.widget.LinearLayout
+import kotlinx.android.synthetic.main.custom_menu.view.*
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt]: WARNING: Wildcard imports should be replaced with imports using fully qualified class names. Wildcard imports can lead to naming conflicts. A library update can introduce naming clashes with your classes which results in compilation errors.

File: app/src/main/res/anim/anim_invisible.xml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<set xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (at-dev-support-01) [inline app/src/main/res/anim/anim_invisible.xml]: WARNING: [UnusedResources] The resource `R.anim.anim_invisible` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/anim/anim_rotate.xml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<set xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (at-dev-support-01) [inline app/src/main/res/anim/anim_rotate.xml]: WARNING: [UnusedResources] The resource `R.anim.anim_rotate` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/anim/anim_visible.xml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<set xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (at-dev-support-01) [inline app/src/main/res/anim/anim_visible.xml]: WARNING: [UnusedResources] The resource `R.anim.anim_visible` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/drawable/custom_item_menu_button.xml [hunk 8]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (at-dev-support-01) [inline app/src/main/res/drawable/custom_item_menu_button.xml]: WARNING: [UnusedResources] The resource `R.drawable.custom_item_menu_button` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/layout/custom_menu.xml [hunk 9]
```diff
@@ -0,0 +1,116 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (at-dev-support-01) [inline app/src/main/res/layout/custom_menu.xml]: WARNING: [UnusedResources] The resource `R.layout.custom_menu` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/drawable/custom_menu_button.xml [hunk 10]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (at-dev-support-01) [inline app/src/main/res/drawable/custom_menu_button.xml]: WARNING: [UnusedResources] The resource `R.drawable.custom_menu_button` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/drawable/custom_textview.xml [hunk 11]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (at-dev-support-01) [inline app/src/main/res/drawable/custom_textview.xml]: WARNING: [UnusedResources] The resource `R.drawable.custom_textview` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+     */
+    internal fun setOnMenuItemClickListener(listener: OnMenuClickListener) {
+        mOnMenuClickListener = listener
+    }
+
+    /**
+     * Interface menu click listener
+     */
+    internal interface OnMenuClickListener {
+        // Event when button share clicked
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+        fun onShareClick()
+
+        // Event when button profile clicked
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+        fun onProfileClick()
+
+        // Event when button calendar clicked
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+import kotlinx.android.synthetic.main.custom_menu.view.*
+import vn.asiantech.way.R
+
+/**
+ * Custom floating button menu
+ * Created by haingoq on 29/09/2017.
+ */
+class MenuLinearLayout(context: Context, attrs: AttributeSet) :
+        LinearLayout(context, attrs), View.OnClickListener {
+    private var mOnMenuClickListener: OnMenuClickListener? = null
```
Comment (at-quocnguyen) [inline app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt]: move to below the `init`

File: app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+import android.view.animation.AnimationUtils
+import android.widget.LinearLayout
+import kotlinx.android.synthetic.main.custom_menu.view.*
+import vn.asiantech.way.R
+
+/**
+ * Custom floating button menu
+ * Created by haingoq on 29/09/2017.
+ */
+class MenuLinearLayout(context: Context, attrs: AttributeSet) :
```
Comment (at-quocnguyen) [inline app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt]: check null với thêm thằng `@JvmOverloads constructor` viết như thế này nèk
```kotlin
class MenuLinearLayout @JvmOverloads constructor(
        context: Context, attrs: AttributeSet? = null) :
        LinearLayout(context, attrs), View.OnClickListener {

    init {
        inflate(context, R.layout.custom_menu, this)
        imgBtnMenu.setOnClickListener(this)
        imgBtnShare.setOnClickListener(this)
        imgBtnProfile.setOnClickListener(this)
        imgBtnCalendar.setOnClickListener(this)
    }

    private var mOnMenuClickListener: OnMenuClickListener? = null
```
https://ask.ericlin.info/post/004-tricks-that-kotlin-can-help-to-write-andorid-custom-views/ đọc cái này để hiểu thêm hehe

Comment (at-quocnguyen) [inline app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt]: với cái tên này nghe kỳ kỳ a nghĩ nên đổi thành `FloatingButtonHorizontal` nge tường minh hơn 😄 smile

File: app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt [hunk 17]
```diff
@@ -0,0 +1,108 @@
+package vn.asiantech.way.ui.views
```
Comment (at-quocnguyen) [inline app/src/main/java/vn/asiantech/way/ui/views/MenuLinearLayout.kt]: chuyển cái này thành package `custom` nha đừng để `views` cho tường minh lun

File: app/src/main/res/layout/custom_menu.xml [hunk 18]
```diff
@@ -0,0 +1,115 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (at-quocnguyen) [inline app/src/main/res/layout/custom_menu.xml]: change name `xml` lun nè --> `custom_floating_button_menu_horizontal`

File: app/src/main/res/values/strings.xml [hunk 19]
```diff
@@ -7,6 +7,11 @@
     <string name="splash_button_enable_location">Enable location</string>
     <string name="splash_toast_turn_on_wifi">Please turn on your internet</string>
 
+    <!--Custom floating menu-->
+    <string name="calendar_title">Calendar</string>
```
Comment (at-quocnguyen) [inline app/src/main/res/values/strings.xml]: tránh nhầm lẫn nên thêm dùm a tên của string vd `custom_floating_menu_...`

**Pull request merged**